app/tomwes/TOMWES_task_4.c

```diff
@@ -0,0 +1,136 @@
+/**
+ * Responsible: Tomasz Wesolowski
+ *
+ *
+ * History log:
+ * 2017.12.07 TOMWES Initial version
+ */
+#include "TOMWES_task_4.h"
+
+void TOMWES_vInit(){
+    
+    int i;
+    for(i=0; i<BUTTON_COUNT*2; i++){
+        asButtonQueue[i].iSize = 1;
+        asButtonQueue[i].iInside = 0;
+        asButtonQueue[i].pQueue = (func_ptr_t*) malloc(sizeof(func_ptr_t));
+    }
+    memset(aiPrevious,-1, BUTTON_COUNT * sizeof(aiPrevious[0]));
+
+}
+
+int iButtonNumber(int i){
```

Comment:
Use types and proper names. 1) Return type in is ok because PIN is int. 2) Input argument is button so type is incorrect. 3) Function could be named `buttonToPin` or `mapButtonToPin`